lib/adapter/TingClientRequestAdapter.php

```diff
@@ -69,7 +69,13 @@ public function execute(TingClientRequest $request) {
         $stopTime = explode(' ', microtime());
         $time = floatval(($stopTime[1]+$stopTime[0]) - ($startTime[1]+$startTime[0]));
 
-        $this->logger->log('Completed SOAP request ' . $soapAction . ' ' . $request->getWsdlUrl() . ' (' . round($time, 3) . 's). Request body: ' . $client->requestBodyString . ' Response: ' . $response);
+        $requestBodyString = $client->requestBodyString;
+        If ($request instanceof TingClientInfomediaRequest ) {
+          // We can't log the request because it contains user data
+          $requestBodyString = '';
+        }
```

Comment:
Instead of stripping the entire body I would prefer a solution where we identify sensitive data e.g. based on parameter names and replace their current values with a placeholder [like we do with Alma](https://github.com/reload/ding2/blob/master/modules/alma/lib/AlmaClient/AlmaClient.class.php#L128-L163) in the logged output.

You might expand this idea to each request type (class) being able to tell exactly which parameter names contain sensitive data. The default would be none.